src/jit/hwintrinsicxarch.cpp

```diff
@@ -905,6 +905,16 @@ GenTree* Compiler::impSSEIntrinsic(NamedIntrinsic        intrinsic,
             break;
         }
 
+        case NI_SSE_StaticCast:
+        {
+            // We fold away the static cast here, as it only exists to satisfy
+            // the type system. It is safe to do this here since the retNode type
+            // and the signature return type are both TYP_SIMD16.
+            assert(sig->numArgs == 1);
+            retNode = impPopStack().val;
```

Comment:
Does this change handle the nested `StaticCast`?

Reply:
Any nested `StaticCast` should have already been handled by this point.

Reply:
But `impPopStack().val` seems not to work with SIMD types. Shall we use `impSIMDPopStack(TYP_SIMDxx)`?

Reply:
Was there a reason that you decided not to use `impSIMDPopStack`?

Reply:
Just an oversight. Fixing now, will push an update shortly (after build completes locally).

Also, to address @fiigii's comment:
> But impPopStack().val seems not working with SIMD types.

`impPopStack()` works just fine with SIMD types, and is what `impSIMDPopStack` uses under the hood. The benefit of using `impSIMDPopStack` is that it performs additional validation on the type and, in certain cases, normalization of the type for existing nodes.

Reply:
> impPopStack() works just fine with SIMD types

In ordinary situation (i.e., on local stack), it indeed works fine. But, IIRC, `impSIMDPopStack` is necessary for some complex environment, such as passing/returning or referred SIMD variables.